Benchmarks_on_OVS_integration_approaches.md

```diff
@@ -0,0 +1,213 @@
+# Benchmarks on OVS integration approaches
+
+OpenStack used to use the legacy Nova-network to provide a basic flat network model. However in its successor Neutron, this model is extended as a flat network cannot meet the requirements of a modern data center. Neutron uses a single bridge called the “integration bridge” that all guest instances are connected to provide connectivity to an external network with a physical network interface. This is essentially the same as the nova-network used to connect VMs to a physical interface.
+
+In QEMU, a L2 network interface is simulated with a tap device. However, in XenServer, vif is more frequently used because of its performance is much better. As Neutron heavily depends on tap device, it seems to be a conflict for the integration between XenServer and OpenStack. [XenAPI: OVS agent updates the wrong port with Neutron](https://review.openstack.org/#/c/242846/) introduces a new bridge called "interim bridge" to work between the guest vm and the integration bridge so vif will still be used while without invasively changing Neutron. This blog post is aiming to investigate the performance cost of "interim bridge" by setting up an environment comparing the conditions with and without interim bridge.
+
+# Terminology Reference
+
+* 'TAP device (tap)': A virtual network kernel device emulates a link layer device and it operates with L2 packets.
+* 'Virtual Network Interfaces (vif)': In XenServer networking, Virtual Network Interfaces emulates network devices in paravirtualized (PV) way, which means guest OS kernel is aware it is running on a virtual machine and it doesn't have to fully emulate a real network device so network communications will be fast and efficient.
+* 'Patch ports': A pair of virtual devices that act as a patch cable to connect multiple Open vSwitch (OVS) bridges (e.g. the integration bridge and interim bridge)
+* 'ovs-vsctl': A CLI high-level tool to for querying and configuring Open vSwitch daemon. Everything we will do to Open vSwitch will be performed by ovs-vsctl.
+* 'iperf': A commonly-used network testing tool that can create TCP/UDP data streams and measure the throughput of a network. iperf works at a Client/Server mode.
+* 'gnuplot': A command-line program can generate plot diagrams.
+
+# Environment setup
+
+![interim bridge](interim.png?raw=true)
+
+We follow below steps to set up two environments like above in a fresh XenServer host.
+
++ Firstly create two private networks serve as “integration bridge” and "interim bridge"
```

Comment:
Instead of calling it an interim bridge, can we call it a "VIF specific bridge" or something else like that?  "Interim bridge" implies that there might just be one of them for all VIFs, but there is just one VIF (with one vif device and one tap device attached)